service/frontend/workflowHandler.go

```diff
@@ -447,14 +447,14 @@ func (wh *WorkflowHandler) PollForActivityTask(
 		return nil, wh.error(errDomainNotSet, scope)
 	}
 
-	if !wh.validIDLength(pollRequest.GetDomain(), scope) {
+	if !wh.validIDLength(pollRequest.GetDomain(), scope, pollRequest.GetDomain()) {
```

Comment:
I would expect the check is only enforced upon creation (i.e. start workflow/create domain, etc)? Otherwise there's nothing user can do to fix this?

Reply:
We do check on workflow start. The check in PollActivityTask should never fail. We can delete all nop checks if you want. But I don't see a very compelling reason to change it from how its already been.

Reply:
I think I have a slight preference for having validation logic in all APIs instead of having APIs have their validation logic depend on which other APIs we know have already run. I find this simpler.

Reply:
ok. For now let's add a comment in all the non-start method and leave the changes there.

Reply:
sounds good - will do.